index.js

```diff
@@ -342,6 +342,25 @@ module.exports = {
     },
 
     /**
+     * Call this if you plan on loading TypeScript files.
+     *
+     * Encore.configureTypeScript(function(tsConfig) {
+     *      // change the tsConfig
+     * });
+     *
+     * Supported configuration options:
+     * @see https://github.com/TypeStrong/ts-loader/blob/master/README.md#available-options
+     *
+     * @param {function} callback
+     * @return {exports}
+     */
+    configureTypeScript(callback) {
+        webpackConfig.configureTypeScript(callback);
```

Comment:
Sorry to go back and forth, but I was thinking more about this - the callback is more difficult to understand and quite a bit harder to use (especially if you want to add a bunch of options at once - you can't just copy some `{foo: bar, baz: true}` code from somewhere and paste it in). I now think we should have this as a normal options object (like you had before). If we eventually add more default options and merging gets weird, we could deprecate the options object and force a callback.